install/azure.go

```diff
@@ -193,7 +193,7 @@ func (k *K8sInstaller) azureSetupServicePrincipal() error {
 		}
 	}
 
-	if k.params.Azure.TenantID == "" && k.params.Azure.ClientID == "" && k.params.Azure.ClientSecret == "" {
+	if k.params.Azure.TenantID == "" && k.params.Azure.ClientID == "" && k.params.Azure.ClientSecret == "" && !k.params.IsDryRun() {
```

Comment:
nit: no directly part of this pr. Would be nice to define a helper func here ie needsAzServicePrincipal(params) bool.